sidecar/src/operator.rs

```diff
@@ -0,0 +1,68 @@
+#![allow(dead_code)] // TODO remove when it is implemented
+
+use anyhow::Result;
+use tracing::{debug, error, info, warn};
+
+use crate::config::Config;
+use crate::controller::Controller;
+use crate::controller::Error;
+
+/// Sidecar operator
+#[derive(Debug)]
+pub struct Operator {
+    /// Operator config
+    config: Config,
+}
+
+impl Operator {
+    /// Constructor
+    #[must_use]
+    #[inline]
+    pub fn new(config: Config) -> Self {
+        Self { config }
+    }
+
+    /// Run operator
+    ///
+    /// # Errors
+    ///
+    /// Return Err when run failed
+    #[inline]
+    pub async fn run(&self) -> Result<()> {
+        let (graceful_tx, mut graceful_rx) = tokio::sync::oneshot::channel();
+        let forceful_shutdown = async {
+            info!("press ctrl+c to shut down gracefully");
+            let _ctrl_c = tokio::signal::ctrl_c().await;
+            let _r = graceful_tx.send(());
+            info!("graceful shutdown already requested, press ctrl+c again to force shut down");
+            let _ctrl_c_c = tokio::signal::ctrl_c().await;
```

Comment:
Why should we ctrl-c twice to shutdown an operator?

Reply:
Pressing Ctrl + C twice forces a shutdown. This is useful when the user knows what they are doing and when a graceful shutdown takes too long to respond.